packages/financial-templates-lib/test/clients/OptimisticOracleEventClient.js

```diff
@@ -0,0 +1,450 @@
+  const identifier = web3.utils.utf8ToHex("Test Identifier");
+
+  const pushPrice = async price => {
+    const [lastQuery] = (await mockOracle.getPendingQueries()).slice(-1);
+    await mockOracle.pushPrice(lastQuery.identifier, lastQuery.time, lastQuery.ancillaryData, price);
+  };
+
+  before(async function() {
+    finder = await Finder.new();
+    timer = await Timer.new();
+
+    // Whitelist an initial identifier we can use to make default price requests.
+    identifierWhitelist = await IdentifierWhitelist.new();
+    await identifierWhitelist.addSupportedIdentifier(identifier);
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.IdentifierWhitelist), identifierWhitelist.address);
+
+    collateralWhitelist = await AddressWhitelist.new();
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.CollateralWhitelist), collateralWhitelist.address);
+
+    store = await Store.new({ rawValue: "0" }, { rawValue: "0" }, timer.address);
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.Store), store.address);
+
+    mockOracle = await MockOracle.new(finder.address, timer.address);
+    await finder.changeImplementationAddress(utf8ToHex(interfaceName.Oracle), mockOracle.address);
+  });
+
+  let requestTxn1, requestTxn2, proposalTxn1, proposalTxn2, disputeTxn1, disputeTxn2, settlementTxn1, settlementTxn2;
+  beforeEach(async function() {
+    // Deploy and whitelist a new collateral currency that we will use to pay oracle fees.
+    collateral = await Token.new("Wrapped Ether", "WETH", 18);
+    await collateral.addMember(1, owner);
+    await collateral.mint(owner, initialUserBalance);
+    await collateral.mint(proposer, initialUserBalance);
+    await collateral.mint(requester, initialUserBalance);
+    await collateral.mint(disputer, initialUserBalance);
+    await collateralWhitelist.addToWhitelist(collateral.address);
+
+    // Set a non-0 final fee for the collateral currency.
+    await store.setFinalFee(collateral.address, { rawValue: finalFee });
+
+    optimisticOracle = await OptimisticOracle.new(liveness, finder.address, timer.address);
+
+    // Contract used to make price requests
+    optimisticRequester = await OptimisticRequesterTest.new(optimisticOracle.address);
+
+    startTime = (await optimisticOracle.getCurrentTime()).toNumber();
+    requestTime = startTime - 10;
+
+    // The Event client does not emit any info `level` events.  Therefore no need to test Winston outputs.
+    // DummyLogger will not print anything to console as only capture `info` level events.
+    dummyLogger = winston.createLogger({
+      level: "info",
+      transports: [new winston.transports.Console()]
+    });
+
+    client = new OptimisticOracleEventClient(
+      dummyLogger,
+      OptimisticOracle.abi,
+      web3,
+      optimisticOracle.address,
+      0, // startingBlockNumber
+      null // endingBlockNumber
+    );
+
+    // Make price requests
```

Comment:
Would we want to generalise this more to consider ancillary data events? in particular, I can still think we'd want to see these logs as we scale out this part of the new contracts.

Reply:
Do you mean generalize the tests to test non trivial ancillary data ?

Reply:
For each individual event function I do test adding an additional event with non trivial ancilllary data, such as [this one](https://github.com/UMAprotocol/protocol/pull/2597/files#diff-d591d6ed4444a39db10d3d068bbca1e2f4a7e01fd14cb5e649d731c5c62cd55bR197) for RequestPrice

Reply:
aha. I see. that looks good to me!